test/e2e_node/kubelet_test.go

```diff
@@ -116,6 +117,64 @@ var _ = framework.KubeDescribe("Kubelet", func() {
 			Expect(err).To(BeNil(), fmt.Sprintf("Error deleting Pod %v", err))
 		})
 	})
+	Context("when scheduling a busybox Pod with hostAliases", func() {
+		podName := "busybox-host-aliases" + string(uuid.NewUUID())
+		expectedHostsFileContent := `# Kubernetes-managed hosts file.
+127.0.0.1	localhost
```

Comment:
there's currently no e2e test checking any other content of the hosts file. so i figure i can help by checking the whole thing.

fixed to just check hostAlias-related entries